File: vihChatBot/src/main/java/com/vihmessenger/vihchatbot/data/repository/EnterprisesDiscoverRepository.kt
package com.vihmessenger.vihchatbot.data.repository

import BaseActivity
import com.vihmessenger.vihchatbot.data.model.EnterPriseDiscoverModel
import com.vihmessenger.vihchatbot.data.model.IndustryResponse
import com.vihmessenger.vihchatbot.data.services.ApiService

class EnterprisesDiscoverRepository(
    private val apiService: ApiService, private val baseActivity: BaseActivity?
) : BaseRepository(baseActivity, apiService) {

    suspend fun getEnterpriseDiscoverListResponse(
        showBlockingLoader: Boolean,
        hashcode: String,
        page: Int,
        search: String,
        industries: String
    ): EnterPriseDiscoverModel {
        return doSafeAPIRequest(
            call = {
                apiService.getEnterpriseDiscoverListResponse(
                    hashcode,
                    page,
                    search,
                    industries
                )
            },
            showBlockingLoader = showBlockingLoader
        ) ?: throw NoConnectionException("Failed to get enterprise discover list")
    }

    suspend fun getIndustriesResponse(
        showBlockingLoader: Boolean
    ): IndustryResponse {
        return doSafeAPIRequest(
            call = { apiService.getIndustries() },
            showBlockingLoader = showBlockingLoader
        ) ?: throw NoConnectionException("Failed to get industries")
    }
}

File: Sources/VihChatBotSDK/Repositories/EnterprisesDiscoverRepository.swift
import Foundation

/// Mirrors `data/repository/EnterprisesDiscoverRepository.kt`.
public final class EnterprisesDiscoverRepository: BaseRepository {

    private let apiService: ApiService

    public init(apiService: ApiService, loaderHost: LoaderHost?) {
        self.apiService = apiService
        super.init(loaderHost: loaderHost)
    }

    public func getEnterpriseDiscoverListResponse(
        showBlockingLoader: Bool,
        hashCode: String,
        page: Int,
        search: String,
        industries: String
    ) async throws -> EnterPriseDiscoverModel {
        try await doSafeAPIRequest(showBlockingLoader: showBlockingLoader) {
            try await self.apiService.getEnterpriseDiscoverList(
                hashcode: hashCode, page: page, search: search, industries: industries
            )
        }
    }

    public func getIndustriesResponse(showBlockingLoader: Bool) async throws -> IndustryResponse {
        try await doSafeAPIRequest(showBlockingLoader: showBlockingLoader) {
            try await self.apiService.getIndustries()
        }
    }
}
